src/components/Commits.jsx

```diff
@@ -68,28 +22,26 @@ function formatDate(date) {
   return `${dd}-${mm}-${yyyy}`;
 }
 
-export default function CommitsPage() {
-  const { data: commits, isLoading } = getAllCommits();
-  if (isLoading) return <> Loading... </>;
+export default function Commits() {
+  const { data: commits, isSuccess } = useAllCommitsQuery();
+
+  if (!isSuccess)
+    return <></>;
 
   return (
     <div>
       <h1 className="commits-title"> Recent commits</h1>
       <ul className="commits-list">
-        {commits?.map((commit) => {
-          return (
-            <>
-              <li key={commit.index} className="commits-list__item">
-                <p className="commits-list__item__message">
-                  {commit.message.split('\n')[0]}
-                </p>
-                <p className="commits-list__item__meta">
-                  On <em>{commit.owner}/{commit.repo}</em> by <strong>{commit.author}</strong> ({formatDate(commit.date)} {formatTime(commit.date)})
-                </p>
-              </li>
-            </>
-          );
-        })}
+        {commits.map((commit, index) =>
+          <li key={index} className="commits-list__item">
```

Comment:
does a commit does not have an ID?

Reply:
Octokit does provide it, but it isn't used anywhere in displaying, so it was left out. I didn't find much of a difference between using an index or explicitly passing the id to use as a key.

Reply:
https://blog.anja-stricker.de/why-using-index-as-key-in-reactjs-map-function-can-cause-problems-and-how-to-fix-them#heading-the-problem-with-using-index-as-the-key

You rather want to use ID. This will break when you add more commits (later)